packages/opentelemetry/src/sampler.ts

```diff
@@ -201,18 +217,28 @@ export function wrapSamplingDecision({
   context,
   spanAttributes,
   sampleRand,
+  sampleRateOverride,
 }: {
   decision: SamplingDecision | undefined;
   context: Context;
   spanAttributes: SpanAttributes;
   sampleRand?: number;
+  sampleRateOverride?: number;
 }): SamplingResult {
   let traceState = getBaseTraceState(context, spanAttributes);
 
   if (sampleRand !== undefined) {
     traceState = traceState.set(SENTRY_TRACE_STATE_SAMPLE_RAND, `${sampleRand}`);
   }
 
+  // We will override the propagated sample rate downstream when
+  // - the tracesSampleRate is applied
+  // - the tracesSampler is invoked
+  // Since unsampled OTEL spans (NonRecordingSpans) cannot hold attributes we need to store this on the (trace)context.
+  if (sampleRateOverride) {
```

Comment:
```suggestion
  if (sampleRateOverride != null) {
```

very important, or it will not propagate 0? 😅

Reply:
oof good catch